<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>625</width>
    <height>776</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout_5">
    <item row="0" column="0">
     <widget class="QTabWidget" name="mainTabStack">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="graphicsTab">
       <property name="toolTip">
        <string>Main Window with Live Graphs</string>
       </property>
       <property name="toolTipDuration">
        <number>5</number>
       </property>
       <attribute name="title">
        <string>Graph</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_3">
        <item>
         <layout class="QVBoxLayout" name="graphTabVerticalLayout">
          <item>
           <widget class="QTabWidget" name="graphsTab">
            <property name="tabPosition">
             <enum>QTabWidget::North</enum>
            </property>
            <property name="currentIndex">
             <number>0</number>
            </property>
            <widget class="QWidget" name="flowGraphTab">
             <attribute name="title">
              <string>Flow</string>
             </attribute>
             <layout class="QGridLayout" name="gridLayout_4">
              <item row="0" column="0">
               <widget class="PlotWidget" name="flowGraph"/>
              </item>
             </layout>
            </widget>
            <widget class="QWidget" name="pressureGraphTab">
             <attribute name="title">
              <string>Pressure</string>
             </attribute>
             <layout class="QGridLayout" name="gridLayout_3">
              <item row="0" column="0">
               <widget class="PlotWidget" name="pressureGraph"/>
              </item>
             </layout>
            </widget>
            <widget class="QWidget" name="currentGraphTab">
             <attribute name="title">
              <string>Current</string>
             </attribute>
             <layout class="QGridLayout" name="gridLayout">
              <item row="0" column="0">
               <widget class="PlotWidget" name="currentGraph"/>
              </item>
             </layout>
            </widget>
            <widget class="QWidget" name="voltageGraphTab">
             <attribute name="title">
              <string>Voltage</string>
             </attribute>
             <layout class="QGridLayout" name="gridLayout_2">
              <item row="0" column="0">
               <widget class="PlotWidget" name="voltageGraph"/>
              </item>
             </layout>
            </widget>
           </widget>
          </item>
          <item>
           <layout class="QHBoxLayout" name="loggingLayout">
            <item>
             <widget class="QLineEdit" name="logFileLineEdit"/>
            </item>
            <item>
             <widget class="QPushButton" name="logButton">
              <property name="text">
               <string>Start Logging</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="infoTickerHorizontalLayout">
            <item>
             <layout class="QVBoxLayout" name="descriptionsLayout">
              <item>
               <widget class="QLabel" name="runStatusLabel">
                <property name="text">
                 <string>Run Status</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLabel" name="loggingLabel">
                <property name="text">
                 <string>Log Status</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QVBoxLayout" name="unitsLayout">
              <item>
               <widget class="QLCDNumber" name="currentReadingLCD"/>
              </item>
              <item>
               <widget class="QComboBox" name="unitsComboBox"/>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QVBoxLayout" name="runCalculateVerticalLayout">
              <item>
               <widget class="QPushButton" name="runStopButton">
                <property name="text">
                 <string>Run</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="calculateResultButton">
                <property name="text">
                 <string>Calculate Result</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="sensorSettingsTab">
       <attribute name="title">
        <string>Sensor Settings</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <item>
         <layout class="QVBoxLayout" name="sensorSettingsVerticalLayout">
          <item>
           <widget class="Line" name="sensorSettingsLine">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="flowMeterSettingLabel">
            <property name="autoFillBackground">
             <bool>false</bool>
            </property>
            <property name="styleSheet">
             <string notr="true">background-color: rgb(0, 0, 0);
color: rgb(238, 238, 236);</string>
            </property>
            <property name="text">
             <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;Flow Meter&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item>
           <layout class="QFormLayout" name="flowMeterSettingsFormLayout">
            <item row="0" column="0">
             <layout class="QHBoxLayout" name="resolutionSettingHorizontalLayout">
              <item>
               <widget class="QSpinBox" name="resolutionSettingSpinBox">
                <property name="minimum">
                 <number>9</number>
                </property>
                <property name="maximum">
                 <number>16</number>
                </property>
                <property name="value">
                 <number>16</number>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLabel" name="resolutionSettingLabel">
                <property name="text">
                 <string>bits</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item row="0" column="1">
             <widget class="QLabel" name="flowMeterResolutionSettingLabel">
              <property name="text">
               <string>Resolution</string>
              </property>
              <property name="alignment">
               <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <widget class="QLabel" name="pressureTransducerSettingsLabel">
            <property name="styleSheet">
             <string notr="true">background-color: rgb(0, 0, 0);
color: rgb(238, 238, 236);</string>
            </property>
            <property name="text">
             <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;Pressure Transducer&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item>
           <layout class="QFormLayout" name="pressureTransducerSettingsFormLayout">
            <item row="0" column="0">
             <layout class="QHBoxLayout" name="transducerHighSideButtonHorizontalLayout">
              <item>
               <widget class="QRadioButton" name="leftTransducerRadioButton">
                <property name="toolTip">
                 <string>The LEFT transducer is connected to the high pressure side.</string>
                </property>
                <property name="toolTipDuration">
                 <number>5</number>
                </property>
                <property name="text">
                 <string>Left</string>
                </property>
                <property name="checked">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QRadioButton" name="rightTransducerRadioButton">
                <property name="toolTip">
                 <string>The RIGHT transducer is connected to the high pressure side.</string>
                </property>
                <property name="toolTipDuration">
                 <number>5</number>
                </property>
                <property name="text">
                 <string>Right</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item row="0" column="1">
             <widget class="QLabel" name="highPressureSideSelectionLabel">
              <property name="layoutDirection">
               <enum>Qt::RightToLeft</enum>
              </property>
              <property name="text">
               <string>High Pressure Side</string>
              </property>
              <property name="alignment">
               <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QTabWidget" name="transducerSettingsTab">
          <property name="currentIndex">
           <number>0</number>
          </property>
          <widget class="QWidget" name="leftTransducerSettingsTab">
           <attribute name="title">
            <string>Left Transducer</string>
           </attribute>
           <layout class="QVBoxLayout" name="verticalLayout">
            <item>
             <layout class="QVBoxLayout" name="leftTransducerSettingsVeritcalLayout">
              <item>
               <layout class="QHBoxLayout" name="leftTransducerCalibrationHorizontalLayout">
                <item>
                 <layout class="QFormLayout" name="leftTransducerCalibrationFormLayout">
                  <item row="0" column="0">
                   <widget class="QLineEdit" name="leftTransducerSlopeLineEdit">
                    <property name="toolTip">
                     <string>Slope of left transducer in [pressure units]/mV</string>
                    </property>
                    <property name="toolTipDuration">
                     <number>5</number>
                    </property>
                   </widget>
                  </item>
                  <item row="0" column="1">
                   <widget class="QLabel" name="leftTransducerSlopeLabel">
                    <property name="text">
                     <string>Slope</string>
                    </property>
                   </widget>
                  </item>
                  <item row="1" column="0">
                   <widget class="QLineEdit" name="leftTransducerInterceptLineEdit">
                    <property name="toolTip">
                     <string>Intercept of left transducer in mV</string>
                    </property>
                   </widget>
                  </item>
                  <item row="1" column="1">
                   <widget class="QLabel" name="leftTransducerInerceptLabel">
                    <property name="text">
                     <string>Intercept</string>
                    </property>
                   </widget>
                  </item>
                 </layout>
                </item>
                <item>
                 <layout class="QFormLayout" name="leftTransducerIDFormLayout">
                  <item row="0" column="1">
                   <widget class="QLabel" name="leftTransducerPressureUnitsLabel">
                    <property name="text">
                     <string>Pressure Units</string>
                    </property>
                   </widget>
                  </item>
                  <item row="1" column="0">
                   <widget class="QLineEdit" name="leftTransducerSerialLineEdit">
                    <property name="toolTip">
                     <string>Left transducer serial number</string>
                    </property>
                    <property name="toolTipDuration">
                     <number>5</number>
                    </property>
                   </widget>
                  </item>
                  <item row="1" column="1">
                   <widget class="QLabel" name="leftTransducerSerialLabel">
                    <property name="text">
                     <string>Serial Number</string>
                    </property>
                   </widget>
                  </item>
                  <item row="0" column="0">
                   <widget class="QComboBox" name="leftTransducerUnitComboBox"/>
                  </item>
                 </layout>
                </item>
               </layout>
              </item>
              <item>
               <widget class="QPushButton" name="loadLeftTransducerSettingsButton">
                <property name="text">
                 <string>Load Cal File</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </widget>
          <widget class="QWidget" name="rightTransducerSettingsTab">
           <attribute name="title">
            <string>Right Transducer</string>
           </attribute>
           <layout class="QHBoxLayout" name="horizontalLayout_2">
            <item>
             <layout class="QVBoxLayout" name="rightTransducerSettingsVerticalLayout">
              <item>
               <layout class="QHBoxLayout" name="rightTransducerCalibrationHorizontalLayout">
                <item>
                 <layout class="QFormLayout" name="rightTransducerCalibrationFormLayout">
                  <item row="0" column="0">
                   <widget class="QLineEdit" name="rightTransducerSlopeLineEdit">
                    <property name="toolTip">
                     <string>Slope of right transducer in Pa/mV</string>
                    </property>
                    <property name="toolTipDuration">
                     <number>3</number>
                    </property>
                   </widget>
                  </item>
                  <item row="0" column="1">
                   <widget class="QLabel" name="rightTransducerSlopeLabel">
                    <property name="text">
                     <string>Slope</string>
                    </property>
                   </widget>
                  </item>
                  <item row="1" column="0">
                   <widget class="QLineEdit" name="rightTransducerInterceptLineEdit">
                    <property name="toolTip">
                     <string>Intercept of left transducer in mV</string>
                    </property>
                    <property name="toolTipDuration">
                     <number>3</number>
                    </property>
                   </widget>
                  </item>
                  <item row="1" column="1">
                   <widget class="QLabel" name="rightTransducerInerceptLabel">
                    <property name="text">
                     <string>Intercept</string>
                    </property>
                   </widget>
                  </item>
                 </layout>
                </item>
                <item>
                 <layout class="QFormLayout" name="rightTransducerIDFormLayout">
                  <item row="0" column="1">
                   <widget class="QLabel" name="rightTransducerPressureUnitsLabel">
                    <property name="text">
                     <string>Pressure Units</string>
                    </property>
                   </widget>
                  </item>
                  <item row="1" column="0">
                   <widget class="QLineEdit" name="rightTransducerSerialLineEdit">
                    <property name="toolTip">
                     <string>Right transducer serial number</string>
                    </property>
                    <property name="toolTipDuration">
                     <number>3</number>
                    </property>
                   </widget>
                  </item>
                  <item row="1" column="1">
                   <widget class="QLabel" name="rightTransducerSerialLabel">
                    <property name="text">
                     <string>Serial Number</string>
                    </property>
                   </widget>
                  </item>
                  <item row="0" column="0">
                   <widget class="QComboBox" name="rightTransducerUnitComboBox"/>
                  </item>
                 </layout>
                </item>
               </layout>
              </item>
              <item>
               <widget class="QPushButton" name="loadRightTransducerSettingsButton">
                <property name="text">
                 <string>Load Cal File</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </widget>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="loadSettingsHorizontalLayout">
          <item>
           <widget class="QLineEdit" name="loadFlowReaderSettingsLineEdit"/>
          </item>
          <item>
           <widget class="QPushButton" name="loadFlowReaderSettingsButton">
            <property name="text">
             <string>Load Settings</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QPushButton" name="saveSensorSettingsButton">
          <property name="text">
           <string>Save Settings</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="experimentSettingsTab">
       <attribute name="title">
        <string>Experiment Settings</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_12">
        <item>
         <layout class="QVBoxLayout" name="experimentsSettingsVerticalLayout">
          <item>
           <layout class="QHBoxLayout" name="saveResultsHorizontalLayout">
            <item>
             <widget class="QLineEdit" name="saveResultsNameLineEdit">
              <property name="toolTip">
               <string>The filename for the results</string>
              </property>
              <property name="toolTipDuration">
               <number>3</number>
              </property>
              <property name="text">
               <string>result.csv</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="saveLocationPushButton">
              <property name="text">
               <string>Save Results Location</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QFormLayout" name="experimentSettingsFormLayout">
            <item row="0" column="0">
             <layout class="QVBoxLayout" name="crossSectionVerticalLayout">
              <item>
               <widget class="QLineEdit" name="crossSectionLineEdit"/>
              </item>
              <item>
               <widget class="QComboBox" name="crossSectionUnitsComboBox"/>
              </item>
             </layout>
            </item>
            <item row="0" column="1">
             <widget class="QLabel" name="crossSectionLabel">
              <property name="text">
               <string>Cross-Sectional Area</string>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QLabel" name="tissueThicknessLabel">
              <property name="text">
               <string>Tissue Thickness</string>
              </property>
             </widget>
            </item>
            <item row="2" column="0">
             <layout class="QVBoxLayout" name="flowSetpointVerticalLayout">
              <item>
               <widget class="QLineEdit" name="flowSetpointLineEdit"/>
              </item>
              <item>
               <widget class="QComboBox" name="flowSetpointUnitsComboBox"/>
              </item>
             </layout>
            </item>
            <item row="2" column="1">
             <widget class="QLabel" name="flowSetpointLabel">
              <property name="text">
               <string>Flow Rate Setpoint</string>
              </property>
             </widget>
            </item>
            <item row="3" column="0">
             <widget class="QSpinBox" name="averagingDepthSpinBox">
              <property name="minimum">
               <number>1</number>
              </property>
              <property name="maximum">
               <number>100</number>
              </property>
             </widget>
            </item>
            <item row="3" column="1">
             <layout class="QVBoxLayout" name="averagingVerticalLayout">
              <item>
               <widget class="QLabel" name="flowRateLabel">
                <property name="text">
                 <string>Flow Rate</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLabel" name="averagingDepthLabel">
                <property name="text">
                 <string>Averaging Depth</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item row="4" column="0">
             <layout class="QVBoxLayout" name="errorBoundsVerticalLayout">
              <item>
               <widget class="QLineEdit" name="errorBoundsLineEdit"/>
              </item>
              <item>
               <widget class="QComboBox" name="errorBoundsUnitsComboBox"/>
              </item>
             </layout>
            </item>
            <item row="4" column="1">
             <widget class="QLabel" name="errorBoundsLabel">
              <property name="text">
               <string>Setpoint Error Bounds</string>
              </property>
             </widget>
            </item>
            <item row="5" column="0">
             <layout class="QVBoxLayout" name="lockTimeVerticalLayout">
              <item>
               <widget class="QLineEdit" name="lockTimeLineEdit"/>
              </item>
              <item>
               <widget class="QComboBox" name="lockTimeUnitsComboBox"/>
              </item>
             </layout>
            </item>
            <item row="5" column="1">
             <widget class="QLabel" name="lockTimeLabel">
              <property name="text">
               <string>Lock In Time</string>
              </property>
             </widget>
            </item>
            <item row="1" column="0">
             <layout class="QVBoxLayout" name="tissueThicknessVerticalLayout">
              <item>
               <widget class="QLineEdit" name="tissueThicknessLineEdit"/>
              </item>
              <item>
               <widget class="QComboBox" name="tissueThicknessUnitsComboBox"/>
              </item>
             </layout>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QVBoxLayout" name="finishedOptionVerticalLayout">
            <item>
             <widget class="QCheckBox" name="soundDoneCheckBox">
              <property name="text">
               <string>Play Sound When Done</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QCheckBox" name="ledDoneCheckBox">
              <property name="text">
               <string>Turn On LED When Done</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QCheckBox" name="popUpDoneCheckBox">
              <property name="enabled">
               <bool>true</bool>
              </property>
              <property name="text">
               <string>Pop-Up with Results When Done</string>
              </property>
              <property name="checked">
               <bool>false</bool>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="loadExperimentSettingsHorizontalLayout">
            <item>
             <widget class="QLineEdit" name="loadExperimentSettingsLineEdit"/>
            </item>
            <item>
             <widget class="QPushButton" name="loadExperimentSettingsButton">
              <property name="text">
               <string>Load Settings</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <widget class="QPushButton" name="saveExperimentSettingsButton">
            <property name="text">
             <string>Save Settings</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="calibrationTab">
       <property name="enabled">
        <bool>true</bool>
       </property>
       <attribute name="title">
        <string>Calibration</string>
       </attribute>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>625</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <customwidgets>
  <customwidget>
   <class>PlotWidget</class>
   <extends>QGraphicsView</extends>
   <header>pyqtgraph</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
